checkup.go

```diff
@@ -275,6 +277,15 @@ func (c *Checkup) UnmarshalJSON(b []byte) error {
 				return err
 			}
 			c.Checkers = append(c.Checkers, checker)
+		case "tcp":
+			var checker TCPChecker
+			// Default timeout
+			checker.Timeout = 1 * time.Second
```

Comment:
Hmm, how about not setting the value here in a JSON unmarshal (which is only one execution path) and instead check the value in doChecks() which is guaranteed to be run every time. If c.Timeout is 0, then you can use something like 1s... or maybe 5s? But I don't think we need to set this value on the struct magically.